WindowState/library/src/main/java/com/microsoft/device/dualscreen/windowstate/WindowState.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.device.dualscreen.windowstate
+
+import android.content.res.Configuration
+import android.graphics.Rect
+import androidx.annotation.VisibleForTesting
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+
+/**
+ * Data class that contains foldable and large screen information extracted from the Jetpack
+ * Window Manager library
+ *
+ * @param hasFold: true if window contains a FoldingFeature,
+ * @param isFoldHorizontal: true if window contains a FoldingFeature with a horizontal orientation
+ * @param foldBounds: Rect object that describes the bound of the FoldingFeature
+ * @param foldState: state of the fold, based on state property of FoldingFeature
+ * @param foldSeparates: based on isSeparating property of FoldingFeature
+ * @param foldOccludes: true if FoldingFeature occlusion type is full
+ * @param windowWidth: Dp value of the window width
+ * @param windowHeight: Dp value of the window height
+ */
+data class WindowState(
+    val hasFold: Boolean = false,
+    val isFoldHorizontal: Boolean = false,
+    val foldBounds: Rect = Rect(),
+    val foldState: FoldState = FoldState.FLAT,
+    val foldSeparates: Boolean = false,
+    val foldOccludes: Boolean = false,
+    val windowWidth: Dp = 0.dp,
+    val windowHeight: Dp = 0.dp,
+) {
+    private val foldableFoldSize = when (isFoldHorizontal) {
+        true -> foldBounds.height()
+        false -> foldBounds.width()
+    }
+
+    val foldablePaneWidth: Int = when (isFoldHorizontal) {
+        true -> foldBounds.right
+        false -> foldBounds.left
+    }
+
+    val foldSize: Int = if (hasFold) foldableFoldSize else 0
```

Comment:
Can we add comments to these properties to show that they return pixel values? That way, no one will make my dp mistake again 😸